_examples/real-world-examples/server-sent-events/server/event_handlers.go

```diff
@@ -0,0 +1,167 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/ThreeDotsLabs/watermill/message"
+	"github.com/ThreeDotsLabs/watermill/message/router/middleware"
+	"github.com/ThreeDotsLabs/watermill/pubsub/gochannel"
+)
+
+const (
+	PostCreatedTopic = "post-created"
+	PostUpdatedTopic = "post-updated"
+	FeedUpdatedTopic = "feed-updated"
+)
+
+func SetupMessageRouter(
+	feedsStorage FeedsStorage,
+	logger watermill.LoggerAdapter,
+) (message.Publisher, message.Subscriber, error) {
+	router, err := message.NewRouter(message.RouterConfig{}, logger)
+	if err != nil {
+		return nil, nil, err
+	}
+	router.AddMiddleware(middleware.Recoverer)
+
+	pubsub := gochannel.NewGoChannel(gochannel.Config{}, logger)
+
+	publishEvents := func(ctx context.Context, tags []string) (messages []*message.Message, err error) {
+		defer func() {
+			if err == nil {
+				logger.Info("Updated posts in feeds", nil)
+			} else {
+				logger.Error("Error in handler", err, nil)
+			}
+		}()
+
+		for _, tag := range tags {
+			logger.Info("Producing event", watermill.LogFields{"tag": tag})
+			event := FeedUpdated{
+				Name:       tag,
+				OccurredAt: time.Now().UTC(),
+			}
+
+			payload, err := json.Marshal(event)
+			if err != nil {
+				return nil, err
+			}
+
+			newMessage := message.NewMessage(watermill.NewUUID(), payload)
+
+			messages = append(messages, newMessage)
+		}
+
+		return messages, nil
+	}
+
+	router.AddHandler(
+		"on-post-created",
```

Comment:
maybe something about read model update will be more concrete?

Reply:
c4eba92ae51e6cb3656a235ee014acfb11293212